### ISSUE TITLE
CudnnLSTM returns all Ones(1) after the 10th sequence 

### ISSUE THREAD

**Issue by josephlau** [tensorflow/tensorflow#13622]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.3
- **Python version**: 3.5
- **Bazel version (if compiling from source)**: v1.3.0-rc1-1486-g752dcb6 1.3.0
- **CUDA/cuDNN version**: 8.0/6.0
- **GPU model and memory**: GTX1050Ti
- **Exact command to reproduce**:

### Describe the problem
I tried to use CudnnLSTM to speed up the training, but found it only returns one after the 10th step, following code generate the output.

### Source code / logs
    import tensorflow as tf
    from tensorflow.contrib.cudnn_rnn import CudnnLSTM
    import numpy as np

    np.set_printoptions(linewidth=240, edgeitems=6)
    # Reset default graph
    tf.reset_default_graph()

    num_layer = 5
    num_unit = 256
    input_size = 400
    seq_lenght = 20

    with tf.device('/gpu:0'):
        x = tf.random_uniform([seq_lenght, input_size], maxval=1, dtype=tf.float32)
        x1 = tf.expand_dims(x, 1)
        lstm = CudnnLSTM(num_layers=num_layer, num_units=num_unit, input_size=input_size,
                     input_mode='linear_input',
                     direction='unidirectional')

        # CudnnLSTM parameter
        lstm_para_size = lstm.params_size()
        lstm_para = tf.Variable(tf.random_uniform([lstm_para_size]), validate_shape=False, name='lstm_para')

        state_c = tf.Variable(tf.zeros(shape=[num_layer, 1, num_unit]), trainable=False)
        state_h = tf.Variable(tf.zeros(shape=[num_layer, 1, num_unit]), trainable=False)

        lstm_output, lstm_h, lstm_c = lstm(input_data=x1, input_h=state_h, input_c=state_c, params=lstm_para)

    # Variable initializing op
    init = tf.global_variables_initializer()

    with tf.Session() as sess:
        sess.run(init)
    cudnn_output = sess.run(lstm_output)
    print(cudnn_output)

###LSTM output
[[[ 0.76159418  0.76159418  0.76159418  0.76159418  0.76159418  0.76159418 ...,  0.76159418  0.76159418  0.76159418  0.76159418  0.76159418  0.76159418]]

 [[ 0.96402758  0.96402758  0.96402758  0.96402758  0.96402758  0.96402758 ...,  0.96402758  0.96402758  0.96402758  0.96402758  0.96402758  0.96402758]]

 [[ 0.99505478  0.99505478  0.99505478  0.99505478  0.99505478  0.99505478 ...,  0.99505478  0.99505478  0.99505478  0.99505478  0.99505478  0.99505478]]

 [[ 0.99932933  0.99932933  0.99932933  0.99932933  0.99932933  0.99932933 ...,  0.99932933  0.99932933  0.99932933  0.99932933  0.99932933  0.99932933]]

 [[ 0.99990922  0.99990922  0.99990922  0.99990922  0.99990922  0.99990922 ...,  0.99990922  0.99990922  0.99990922  0.99990922  0.99990922  0.99990922]]

 [[ 0.99998772  0.99998772  0.99998772  0.99998772  0.99998772  0.99998772 ...,  0.99998772  0.99998772  0.99998772  0.99998772  0.99998772  0.99998772]]

 ..., 
 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]

 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]

 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]

 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]

 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]

 [[ 1.          1.          1.          1.          1.          1.         ...,  1.          1.          1.          1.          1.          1.        ]]]

**Comment by tatatodd** [comment]
@zheng-xq Can you comment on this, or suggest someone who can?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by zheng-xq** [comment]
@protoget, could you take a look at this? Thanks!

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by protoget** [comment]
We have introduced new CudnnLSTM layers after TF1.5 which manages the variables for you. Please try that out.
Also, note that the code doesn't show how train_ops are created.
```py
    lstm_para_size = lstm.params_size()
    lstm_para = tf.Variable(tf.random_uniform([lstm_para_size]), validate_shape=False, name='lstm_para')

    state_c = tf.Variable(tf.zeros(shape=[num_layer, 1, num_unit]), trainable=False)
    state_h = tf.Variable(tf.zeros(shape=[num_layer, 1, num_unit]), trainable=False)

    lstm_output, lstm_h, lstm_c = lstm(input_data=x1, input_h=state_h, input_c=state_c, params=lstm_para)
```

If you're just doing inference, does "1" show up in one step or persists after one step? I suspect with your input it happens that the ground truth output is 1 for certain step.

**Comment by xingjinglu** [comment]
@protoget  I find it difficult to user new CudnnLSTM layer api in TF1.5, are there some sample use case for the new api?

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by protoget** [comment]
@xingjinglu I think the new CudnnLSTM **layer** should make things way simpler.
you can check out unittest of the class to see examples.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/cudnn_rnn/python/kernel_tests/cudnn_rnn_test.py#L118
